pkg/envoy/envoy_container_logs.go

```diff
@@ -0,0 +1,68 @@
+package envoy
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"regexp"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/openservicemesh/osm-health/pkg/common"
+)
+
+// Verify interface compliance
+var _ common.Runnable = (*NoBadEnvoyLogsCheck)(nil)
+
+// NoBadEnvoyLogsCheck implements common.Runnable
+type NoBadEnvoyLogsCheck struct {
+	client kubernetes.Interface
+	pod    *v1.Pod
+}
+
+// HasNoBadEnvoyLogsCheck checks whether the envoy container of the pod has bad/error/warning log messages
+func HasNoBadEnvoyLogsCheck(client kubernetes.Interface, pod *v1.Pod) NoBadEnvoyLogsCheck {
+	return NoBadEnvoyLogsCheck{
+		client: client,
+		pod:    pod,
+	}
+}
+
+// Info implements common.Runnable
+func (check NoBadEnvoyLogsCheck) Info() string {
+	return fmt.Sprintf("Checking whether pod %s has bad/error logs in envoy container", check.pod.Name)
+}
+
+// Run implements common.Runnable
+func (check NoBadEnvoyLogsCheck) Run() error {
+	podLogsOpt := v1.PodLogOptions{
+		Container: "envoy",
+		Follow:    true,  // Follow the log stream of the pod.
+		Previous:  false, // Return previously terminated container logs.
+	}
+
+	request := check.client.CoreV1().Pods(check.pod.Namespace).GetLogs(check.pod.Name, &podLogsOpt)
+	envoyPodLogsReader, err := request.Stream(context.TODO())
+	if err != nil {
+		return fmt.Errorf("could not stream envoy container logs of pod %s: %#v", check.pod.Name, err)
+	}
+	defer envoyPodLogsReader.Close() //nolint: errcheck,gosec
+
+	re := regexp.MustCompile("(?i)(error)|(warn)")
+	scanner := bufio.NewScanner(envoyPodLogsReader)
+	lines := 20
+	for scanner.Scan() {
+		logLine := scanner.Text()
+		log.Info().Msg(logLine)
+		lines--
+		if lines == 0 {
+			return nil
+		}
+		if re.MatchString(logLine) {
+			// take note of the error
+		}
+	}
```

Comment:
@draychev Scanning each line of the envoy container log (whether using `string.Contains` or `regexp` is too slow).  Even copying the entire `envoyPodLogsReader` reader to a string variable takes so much time because envoy logs are very verbose.

Reply:
Restricted the number of logs through `TailLines` in pod log options. I chose to process the last 1000 logs.